wizeline/wize-q-remix
# feat: add manager to department.

#### What does this PR do?
- add the field `manager_employee_id`
- add the relation between users and department
- Add the UI of the admin panel to show the departments 
- Add the UI to assign an admin to a department 
- Add the UI to add users to a department


#### How should this be manually tested?
1. Login as an admin.
2. Go to the panel admin
3. Select de department tab
4. Select the option to edit a department
5. Add an admin and save. 
6. Double check that the user selected appears as manager
7. Select the option to see the users.
8. Verify that you can see the users assigned to the department selected.
9. Remove one user and double check that the user disappears.
10. Add a user using the input search and add the user selected. 
11. Verify that the pagination and the search are working as expected. 

#### Any background context you want to provide?
As part of the new changes, we have to add the functionality to have a manager for each department.
For futures changes, we are going to use this manager to send an email by default when someone selects a department when creating a question.   

#### What are the relevant tickets?
Closes #205 

#### Screenshots
:camera:

[admin-panel]
<img width="1788" alt="Screen Shot 2023-04-21 at 17 48 40" src="https://user-images.githubusercontent.com/97203617/233750023-878721ae-0a1c-41c7-badd-86b683e71f30.png">

[Department tab]
<img width="1788" alt="Screen Shot 2023-04-21 at 17 52 15" src="https://user-images.githubusercontent.com/97203617/233750080-f0f88e00-4fc4-48ad-9169-d0a794478f01.png">

[Edit Department]
<img width="520" alt="Screen Shot 2023-04-21 at 17 53 22" src="https://user-
images.githubusercontent.com/97203617/233750145-eabc3c59-8b56-4542-9aab-f7bc6d0dc750.png">

[User by department]

<img width="487" alt="Screen Shot 2023-04-21 at 17 54 18" src="https://user-images.githubusercontent.com/97203617/233750218-6590bbcf-dd99-4edd-b116-e9f10506b409.png">


https://user-images.githubusercontent.com/97203617/233750790-e7dc791a-2b43-4ac2-9e39-7f498536d756.mp4
